### PR TITLE
Deprecate service point for server certificate

### PR DESCRIPTION
**Pull request by aik-jahoda**
Service point no longer supports server certificate. This PR link to HttpClient including C# example

Partially fixes dotnet/runtime#33125

fixes dotnet/runtime#24773

### CODE REVIEW

File: xml/System.Net/ServicePoint.xml [hunk 1]
```diff
@@ -204,7 +205,9 @@
 ## Remarks  
  Although a <xref:System.Net.ServicePoint> object can make multiple connections to an Internet resource, it can maintain only one certificate.  
   
-   
+> [!NOTE]
+>  From .Net Core 1.0 and later including .NET 5.0 and later, the <xref:System.Net.ServicePoint.Certificate%2A> property returns allways null. The <xref:System.Net.Http.HttpClientHandler.ServerCertificateCustomValidationCallback%2A> is recomended way how to access the server certificate.
```
Comment (Youssef1313) [inline xml/System.Net/ServicePoint.xml:209]:
```suggestion
> Starting in .NET Core 1.0 and later including .NET 5.0 and later, the <xref:System.Net.ServicePoint.Certificate%2A> property always returns null. The <xref:System.Net.Http.HttpClientHandler.ServerCertificateCustomValidationCallback%2A> is the recomendded way to access the server certificate.
```

File: samples/snippets/csharp/VS_Snippets_Misc/system.net.http.httpclienthandler-secure/cs/program.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+using System.IO;
+using System.Net;
+using System.Net.Http;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+
+class HttpClientHandler_SecureExample
+{
+// <Snippet1>
```
Comment (Youssef1313) [inline samples/snippets/csharp/VS_Snippets_Misc/system.net.http.httpclienthandler-secure/cs/program.cs:11]:
```suggestion
    // <Snippet1>
```

File: samples/snippets/csharp/VS_Snippets_Misc/system.net.http.httpclienthandler-secure/cs/program.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    static async Task Main()
+    {
+        // Create an HttpClientHandler object and set to use default credentials
+        HttpClientHandler handler = new HttpClientHandler();
```
Comment (Youssef1313) [inline samples/snippets/csharp/VS_Snippets_Misc/system.net.http.httpclienthandler-secure/cs/program.cs:15]:
```suggestion
        var handler = new HttpClientHandler();
```

File: samples/snippets/csharp/VS_Snippets_Misc/system.net.http.httpclienthandler-secure/cs/program.cs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+        // Set custom server validation callback
+        handler.ServerCertificateCustomValidationCallback = ServerCertificateCustomValidation;
+
+        // Create an HttpClient object
+        HttpClient client = new HttpClient(handler);
```
Comment (Youssef1313) [inline samples/snippets/csharp/VS_Snippets_Misc/system.net.http.httpclienthandler-secure/cs/program.cs:21]:
```suggestion
        var client = new HttpClient(handler);
```

File: samples/snippets/csharp/VS_Snippets_Misc/system.net.http.httpclienthandler-secure/cs/program.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        try
+        {
+            HttpResponseMessage response = await client.GetAsync("https://docs.microsoft.com/");
+
+            response.EnsureSuccessStatusCode();
+
+            string responseBody = await response.Content.ReadAsStringAsync();
+            Console.WriteLine($"Read {responseBody.Length} characters");
+        }
+        catch(HttpRequestException e)
```
Comment (Youssef1313) [inline samples/snippets/csharp/VS_Snippets_Misc/system.net.http.httpclienthandler-secure/cs/program.cs:33]:
```suggestion
        catch (HttpRequestException e)
```

File: samples/snippets/csharp/VS_Snippets_Misc/system.net.http.httpclienthandler-secure/cs/program.cs [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        {
+            Console.WriteLine("\nException Caught!");
+            Console.WriteLine("Message :{0} ",e.Message);
```
Comment (Youssef1313) [inline samples/snippets/csharp/VS_Snippets_Misc/system.net.http.httpclienthandler-secure/cs/program.cs:36]:
```suggestion
            Console.WriteLine($"Message: {e.Message} ");
```

File: samples/snippets/csharp/VS_Snippets_Misc/system.net.http.httpclienthandler-secure/cs/program.cs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        Console.WriteLine($"Effective date: {certificate.GetEffectiveDateString()}");
+        Console.WriteLine($"Exp date: {certificate.GetExpirationDateString()}");
+        Console.WriteLine($"Issuer: {certificate.Issuer}");
+        Console.WriteLine($"Subject: {certificate.Subject}");
+
+        // Based on the custom logic it is possible to decide whether the client considers certificate valid or not
+        Console.WriteLine($"Errors: {sslErrors}");
+        return sslErrors == SslPolicyErrors.None;
+    }
+// </Snippet1>
```
Comment (Youssef1313) [inline samples/snippets/csharp/VS_Snippets_Misc/system.net.http.httpclienthandler-secure/cs/program.cs:58]:
```suggestion
    // </Snippet1>
```

File: xml/System.Net.Http/HttpClientHandler.xml [hunk 8]
```diff
@@ -893,7 +893,19 @@ handler.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousA
...
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+
+## Remarks
+ The <xref:System.Net.Http.HttpClientHandler.ServerCertificateCustomValidationCallback%2A> can be used to obtain and validate the server cetrificate.
+
+## Examples
+ The following code example displays the server certificate.
+
+ [!code-csharp[System.Net.Http.HttpClientHandler#5](~1samples/snippets/csharp/VS_Snippets_Misc/system.net.http.httpclienthandler-secure/cs/secure.cs#1)]
```
Comment (Youssef1313) [inline xml/System.Net.Http/HttpClientHandler.xml:905]:
```suggestion
 [!code-csharp[System.Net.Http.HttpClientHandler#5](~samples/snippets/csharp/VS_Snippets_Misc/system.net.http.httpclienthandler-secure/cs/secure.cs#1)]
```

File: xml/System.Net.Http/HttpClientHandler.xml [hunk 9]
```diff
@@ -893,7 +893,19 @@ handler.ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousA
       <Docs>
         <summary>Gets or sets a callback method to validate the server certificate.</summary>
         <value>A callback method to validate the server certificate.</value>
-        <remarks>To be added.</remarks>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+
+## Remarks
+ The <xref:System.Net.Http.HttpClientHandler.ServerCertificateCustomValidationCallback%2A> can be used to obtain and validate the server cetrificate.
```
Comment (ManickaP) [inline xml/System.Net.Http/HttpClientHandler.xml:900]: typo:
```suggestion
 The <xref:System.Net.Http.HttpClientHandler.ServerCertificateCustomValidationCallback%2A> can be used to obtain and validate the server certificate.
```

**Review by ManickaP — APPROVED** [review]
Small typo, otherwise LGTM!

**Pull request merged**